service/src/main/java/uk/gov/hmcts/reform/fpl/service/HearingBookingService.java

```diff
@@ -119,6 +151,99 @@ public HearingBooking getHearingBookingByUUID(List<Element<HearingBooking>> hear
+            .collect(Collectors.toList());
+
+        return newHearings.stream()
+            .filter(newHearing -> !oldHearingIDs.contains(newHearing.getId()))
+            .collect(Collectors.toList());
+    }
+
+    public Map<String, Object> getHearingNoticeCaseFields(
+        List<Element<HearingBooking>> newHearings, List<Element<HearingBooking>> oldHearings) {
+        StringBuilder stringBuilder = new StringBuilder();
+        List<UUID> oldHearingIDs = oldHearings.stream()
+            .map(Element::getId)
+            .collect(Collectors.toList());
+
+        int selectorMinValue = -1;
+        for (int i = 0; i < newHearings.size(); i++) {
+            if (!oldHearingIDs.contains(newHearings.get(i).getId())) {
+                Element<HearingBooking> hearingElement = newHearings.get(i);
+                String newHearingLabel = format("Hearing %d: %s hearing %s", i + 1,
+                    hearingElement.getValue().getType().getLabel(),
+                    formatLocalDateTimeBaseUsingFormat(hearingElement.getValue().getStartDate(), DATE));
+
+                stringBuilder.append(newHearingLabel).append("\n");
+
+                if (selectorMinValue < 0) {
+                    selectorMinValue = i;
+                }
+            }
+        }
+        Selector selector = Selector.newSelector(newHearings.size(), selectorMinValue, newHearings.size());
+
+        return Map.of(NEW_HEARING_LABEL.getKey(), stringBuilder.toString(), NEW_HEARING_SELECTOR.getKey(), selector);
+    }
+
+    public List<Element<HearingBooking>> getSelectedHearings(Selector hearingSelector,
+                                                             List<Element<HearingBooking>> hearings) {
+        if (hearings.isEmpty() || hearingSelector == null) {
+            return List.of();
+        } else {
+            return hearingSelector.getSelected().stream()
+                .map(hearings::get)
+                .collect(toList());
+        }
+    }
+
+    public List<Element<HearingBooking>> getHearingDetails(
+        CaseDetails caseDetails, CaseData caseData, CaseData caseDataBefore) {
+        List<Element<HearingBooking>> hearings = caseData.getHearingDetails();
+        List<Element<HearingBooking>> hearingsBefore = new ArrayList<>(
+            defaultIfNull(caseDataBefore.getHearingDetails(), emptyList()));
+
+        removePastHearings(hearingsBefore);
+
+        if (getNewHearings(hearings, hearingsBefore).isEmpty()) {
+            caseDetails.getData().put(NEW_HEARING_SELECTOR.getKey(), null);
+        }
+
+        List<Element<HearingBooking>> updatedHearings =
+            setHearingJudge(caseData.getHearingDetails(), caseData.getAllocatedJudge());
+
+        sendDocumentsForSelectedHearings(caseDetails, caseData, updatedHearings);
+
+        return combineHearingDetails(updatedHearings,
+            getPastHearings(defaultIfNull(caseDataBefore.getHearingDetails(), emptyList())));
+    }
+
+    private void sendDocumentsForSelectedHearings(
```

Comment:
Not sure if I like "sendDocument" sending where? it uploads to docs store and genarates, I believe better name (or splitting into multiple,would be more beneficial)

Reply:
agree that name is not great. I was trying to refactor out all this code from controller without knowing much about the code.

Reply:
updated name